### PR TITLE
Feature forum message preview

### PR DESCRIPTION
**Pull request by felipecalvo**
## :dart: Goal

Let moderators preview their forum messages before sending.

## :memo: Details

We'll be keeping this to moderators and up for now as talked with @rgonzalezt and @lauramangifesta. We might extend this to students in the future if we add some kind of document or guidelines for Markdown editing.

## :camera_flash: Screenshots
image

____
image

### CODE REVIEW

**Comment by felipecalvo** [comment]
Added icons on buttons.

image
image

File: app/controllers/discussions_messages_controller.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ def destroy
     redirect_back(fallback_location: root_path)
   end
 
+  def preview
+    render json: { preview: Message.new(content: URI.decode(params[:content])).content_html }
```
Comment (luchotc) [inline app/controllers/discussions_messages_controller.rb:19]: Rails does decode GET query params by default, so I'm guessing it's not liking something in the way we are encoding the param in JS. If we can change that to get rid of this, awesome, if not it's ok too :smile:

**Review by luchotc — APPROVED** [review]
This looks https://www.youtube.com/watch?v=dliA3Bdt4eQ :heart:

File: lib/mumuki/laboratory/locales/en.yml [hunk 2]
```diff
@@ -79,6 +79,7 @@ en:
   dont_leave_us: Don't leave us! Learning is fun. You just have to keep at it.
   download: Download your solution
   edit: Edit
+  edit_mensaje: Edit message
```
[the remark this answers is not in the document]
Comment (felipecalvo) [reply lib/mumuki/laboratory/locales/en.yml:82]: No dirty translators were harmed in the making of this commit

File: app/controllers/discussions_messages_controller.rb (hunk 1 above)
Comment (felipecalvo) [reply app/controllers/discussions_messages_controller.rb:19]: I removed both this `URI.decode` in Ruby and the `encodeURIComponent()` in JS and everything still works. I was just being extra cautious I guess.

I couldn't break it in any case but we might want to try with all kinds of funky characters just to be sure.

**Pull request merged**